From cbdf3171073572918c21dee92a1123af522631cb Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Fri, 6 Sep 2024 15:32:05 +0800
Subject: [PATCH 199/277] mtk: mt76: mt7996: Fix call trace happened when wed
 attach fail

1. when we attech wed0 fail, we will not going to attach wed1

Signed-off-by: Rex Lu <rex.lu@mediatek.com>
---
 mt7996/mmio.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mt7996/mmio.c b/mt7996/mmio.c
index 5066376d..2e975054 100644
--- a/mt7996/mmio.c
+++ b/mt7996/mmio.c
@@ -475,8 +475,11 @@ int mt7996_mmio_wed_init(struct mt7996_dev *dev, void *pdev_ptr,
 		wed->wlan.reset_complete = mt76_wed_reset_complete;
 	}
 
-	if (mtk_wed_device_attach(wed))
+	if (mtk_wed_device_attach(wed)) {
+		wed_enable = false;
+		dev->has_rro = false;
 		return 0;
+	}
 
 	*irq = wed->irq;
 	dev->mt76.dma_dev = wed->dev;
-- 
2.45.2

